test/pstl_offload/memory/usm_memory_replacement.cpp

```diff
@@ -19,6 +19,8 @@
 
 #include "support/utils.h"
 
+#include "system_allocations.h"
```

Comment:
May be it would be better to move this test into the separate test file?
Currently, we use `usm_memory_replacement.pass` as part of subsets in CI and other testings where testing of the memory across TU is not critical.

Reply:
Let's discuss. May be I'm not fully understand what is "critical" here, at least it cheap.

I modified existing test vs added new one, because:
* Additional execution time is low vs compile time for a new test.
* Added checks do exactly what is in the test name, i.e. checks "usm memory replacement".

Reply:
I guess you are right. And also `make usm_memory_replacement.pass` works now, so I am closing this issue.

Reply:
Since `usm_memory_replacement.pass` is part of testing subset on the compiler side, let's provide only single-TU tests to them and hence move this test into separate file.

Reply:
Done, this is sound reason.